src/app/tests/TestReadInteraction.cpp

```diff
@@ -161,12 +174,31 @@ class MockInteractionModelApp : public chip::app::InteractionModelDelegate
     }
 
     bool mGotEventResponse      = false;
+    bool mGotAttributeResponse  = false;
+    bool mGotReport             = false;
     bool mGotReadStatusResponse = false;
 };
 } // namespace
 
 namespace chip {
 namespace app {
+CHIP_ERROR ReadSingleClusterData(ClusterInfo & aClusterInfo, TLV::TLVWriter * apWriter, bool * apDataExists)
+{
+    CHIP_ERROR err   = CHIP_NO_ERROR;
+    uint64_t version = 0;
+    VerifyOrExit(aClusterInfo.mClusterId == kTestClusterId && aClusterInfo.mEndpointId == kTestEndpointId,
+                 err = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(apWriter != nullptr, /* no op */);
```

Comment:
why is a null apWriter a 'CHIP_NO_ERROR' ?

Reply:
update it with CHIP_ERROR_INVALID_ARGUMENT, thanks